prepare-operator-metadata.yml

```diff
@@ -21,20 +21,54 @@
     operator_work_dir: "/home/jenkins/agent/test-operator"
     operator_dir: "/tmp/test"
     work_dir: "{{ lookup('env', 'WORKSPACE') }}"
-    jq_bin_path : "/usr/bin/jq"
+    jq_bin_path: "/usr/bin/jq"
     yq_bin_path: "/usr/local/bin/yq"
     testing_bin_path: "/usr/local/bin"
-    current_channel: '' # Added to avoid a bug with undefined variables
+    current_channel: ''
+    # Added to avoid a bug with undefined variables
 
   tasks:
-    
+
     - name: "Set variables for operator_work_dir and operator_dir"
       set_fact:
-        operator_work_dir: "{{ operator_work_dir | default('/home/jenkins/agent/test-operator')}}"
+        operator_work_dir: "{{ operator_work_dir |
+                               default('/home/jenkins/agent/test-operator')}}"
         operator_dir: "{{ operator_dir | default('/tmp/test') }}"
+        parse_operator_metadata_results: {'result': '', 'msg': ''}
+
+    - block:
+        - name: "Run operator-courier nest to copy operator metadata in nested format to the work dir"
+          shell: "operator-courier nest {{ operator_dir }} {{ operator_work_dir }}"
+
+        - name: "Parse operator metadata needed to run the tests"
+          include_role:
+            name: parse_operator_metadata
+
+      rescue:
+        - name: "Rescue block contains the error messages"
+          debug:
+            msg: "Rescue block has found an error, The following are details of failed task."
+        - name: "FAILED task name in ansible is as follows:"
+          debug:
+            msg: "{{ ansible_failed_task.name }}"
+
+        - name: "Result of failed task"
+          debug:
+            msg: "{{ ansible_failed_result }}"
+
+        - name: "Set failure result"
+          set_fact:
+            parse_operator_metadata_results: {'result': 'fail', 'msg': "{{ ansible_failed_result }}"}
```

Comment:
I am sorry but returning out put of ansible task is not very user-friendly.

Usually, it is very difficult to me to parse output of tasks. Would it be possible to return human friendly description of failure and ideally few words how to fix that.

Reply:
@lslebodn 
No problem. 
More reviews more robust the code. Sure I can add an additional task name inside the results message and why it probably failed.